src/main/java/chess/domain/ChessGame.java

```diff
@@ -0,0 +1,39 @@
+package chess.domain;
+
+import chess.domain.board.Board;
+import chess.domain.board.Square;
+import chess.domain.piece.Piece;
+import java.util.List;
+import java.util.Optional;
+
+public class ChessGame {
+
+    private final Board board = Board.generate();
+
+    public void move(final Square source, final Square destination) {
+        final Optional<Piece> pieceOf = board.findPieceOf(source);
+        if (pieceOf.isEmpty()) {
+            throw new IllegalArgumentException("움직일 기물이 존재하지 않습니다.");
+        }
```

Comment:
```suggestion
        final Piece pieceOf = board.findPieceOf(source).orElseThrow("~");
```

Optional 를 받는 시점부터 처리해볼 수 있을 것 같아요.

Reply:
저런 방법이 있었군요 좋은 방법 알려주셔서 감사합니다 🔥